sycl/include/sycl/ext/oneapi/experimental/root_group.hpp

```diff
@@ -0,0 +1,134 @@
+//==--- root_group.hpp --- SYCL extension for root groups ------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <sycl/builtins.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
+#include <sycl/memory_enums.hpp>
+#include <sycl/queue.hpp>
+
+#define SYCL_EXT_ONEAPI_ROOT_GROUP 1
+
+namespace sycl {
+__SYCL_INLINE_VER_NAMESPACE(_V1) {
+namespace ext::oneapi::experimental {
+
+namespace info::kernel_queue_specific {
+struct max_num_work_group_sync {
+  using return_type = size_t;
+};
+} // namespace info::kernel_queue_specific
```

Comment:
Can you add a TODO here to potentially revisit and align with sycl_ext_oneapi_forward_progress?

https://github.com/intel/llvm/pull/7598 suggests a more generic `max_num_work_groups`, and I suspect we will want to implement that instead.  I'm sorry these extensions have crossed -- I was holding off on updating the root-group proposal until https://github.com/intel/llvm/pull/7598 was reviewed/merged.